library/src/components/shared/ConfirmEditDialog.tsx

```diff
@@ -0,0 +1,103 @@
+import React, {
+  useState,
+  useEffect,
+  forwardRef,
+  ComponentType,
+  FormEvent,
+} from 'react';
+import Dialog from '@mui/material/Dialog';
+import DialogTitle from '@mui/material/DialogTitle';
+import DialogContent from '@mui/material/DialogContent';
+import DialogActions from '@mui/material/DialogActions';
+import Button from '@mui/material/Button';
+import LinearProgress from '@mui/material/LinearProgress';
+import { DialogContentText, Slide, SlideProps } from '@mui/material';
+import CloseRoundedIcon from '@mui/icons-material/CloseRounded';
+import _ from '../../services/translations/translate';
+
+const Transition: ComponentType<any> = forwardRef(function (
+  props: SlideProps,
+  ref: React.Ref<unknown>
+) {
+  return <Slide {...props} direction='up' ref={ref} />;
+});
+Transition.displayName = 'ConfirmDialogTransition';
+
+interface ConfirmEditDialogProps {
+  text: React.ReactNode;
+  open: boolean;
+  formEvent?: FormEvent<HTMLFormElement>;
+  handleClose: () => void;
+  handleSave: (e: FormEvent<HTMLFormElement>) => void;
+}
+
+export const ConfirmEditionDialog = (props: ConfirmEditDialogProps) => {
+  const { open, handleClose, text, handleSave, formEvent } = props;
+  const TOTAL_TIME = 100;
+  const [progress, setProgress] = useState(TOTAL_TIME);
+
+  useEffect(() => {
+    let timer: NodeJS.Timeout | null = null;
+    if (open) {
+      timer = setInterval(() => {
+        setProgress((oldProgress) => {
+          if (oldProgress === 0) {
+            return 0;
+          }
+          return oldProgress - 5;
+        });
+      }, 250);
+    }
+
+    setProgress(TOTAL_TIME);
+
+    return () => {
+      if (timer) {
+        clearInterval(timer);
+      }
+    };
+  }, [open]);
```

Comment:
```suggestion
  useEffect(() => {
    let timer: NodeJS.Timeout | null = null;
    if (open) {
      timer = setInterval(() => {
        setProgress((oldProgress) => {
          if (oldProgress === 0) {
            return 0;
          }
          return oldProgress - 5;
        });
      }, 250);
    }

    setProgress(TOTAL_TIME);

    return () => {
      if (timer) {
        clearInterval(timer);
      }
    };
  }, [open, TOTAL_TIME]);
```
The behavior will be exactly the same and this is what is recommended.

Reply:
TOTAL_TIME is a constant, so, it doesn't have sense to check the possible changes on it

Reply:
right